x-pack/plugins/maps/public/classes/styles/vector/style_fields_custodian.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  CATEGORICAL_DATA_TYPES,
+  FIELD_ORIGIN,
+  ORDINAL_DATA_TYPES,
+  VECTOR_STYLES,
+} from '../../../../common/constants';
+import { IField } from '../../fields/field';
+
+export interface StyleField {
+  label: string;
+  name: string;
+  origin: FIELD_ORIGIN;
+  type: string;
+  supportsAutoDomain: boolean;
+}
+
+export async function styleFieldsCustodianFactory(fields: IField[]): Promise<StyleFieldsCustodian> {
```

Comment:
nit: mind renaming to `createStyleFieldsHelper`. For `factory`, I always think about the very specific OO-pattern.